src/main/java/com/amazonaws/services/dynamodbv2/datamodeling/encryption/DynamoDBEncryptor.java

```diff
@@ -405,16 +418,22 @@ private void actualEncryption(Map<String, AttributeValue> itemAttributes,
                             dk.encrypt(toByteArray(plainText), null, encryptionMode));
                 } else {
                     if (cipher == null) {
+                        blockSize = getBlockSize(encryptionMode);
                         cipher = Cipher.getInstance(encryptionMode);
-                        ivSize = cipher.getBlockSize();
                     }
                     // Encryption format: <iv><ciphertext>
                     // Note a unique iv is generated per attribute
-                    byte[] iv = Utils.getRandom(ivSize);
-                    cipher.init(Cipher.ENCRYPT_MODE, encryptionKey, new IvParameterSpec(iv), Utils.getRng());
-                    cipherText = ByteBuffer.allocate(ivSize + cipher.getOutputSize(plainText.remaining()));
-                    cipherText.put(iv);
+                    cipher.init(Cipher.ENCRYPT_MODE, encryptionKey, Utils.getRng());
+                    cipherText = ByteBuffer.allocate(blockSize + cipher.getOutputSize(plainText.remaining()));
+                    cipherText.position(blockSize);
```

Comment:
Block size may not equal IV size, depending on the mode (e.g. AES-GCM would use a 12 byte IV with a 16 byte block size.

Reply:
In this data-format  version of the client they must be the same. It's a limitation of the format which we cannot change until we do a more significant version rev.

Reply:
Please leave a comment to this effect, and a runtime check to confirm that the returned `iv` array is the correct size.